spec/lib/oauth/client_authentication/none_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Doorkeeper::OAuth::ClientAuthentication::None do
+  describe 'matches_request?' do
+    it "matches if the request doesn't have authorization or client_secret" do
+      request = mock_request({
+        client_id: '1234'
+      })
+
+      expect(described_class.matches_request?(request)).to be true
+    end
+
+    it "doesn't match if the request has client_secret" do
+      request = mock_request({
+        client_id: '1234',
+        client_secret: "5678"
+      })
+
+      expect(described_class.matches_request?(request)).to_not be true
+    end
+
+    it "doesn't match if the request has authorization" do
+      request = mock_request({
+        client_id: '1234'
+      }, ActionController::HttpAuthentication::Basic.encode_credentials('username', 'password'))
+
+      binding.break
+      expect(described_class.matches_request?(request)).to_not be true
+    end
+
+    private
+
+    # I'm not sure if there's a better way to get a mock rack request for
+    # testing. Here we don't need a full request spec, but we do need enough to
+    # check that the logic of these classes works.
+    def mock_request(params, credentials = nil)
```

Comment:
There's gotta be a better way to get a fake request that acts in a given way in ruby, surely?

Reply:
This comment still applies, I just changed the method signature to use keyword args.